modin/pandas/groupby.py

```diff
@@ -930,138 +1022,103 @@ def _wrap_aggregation(
         ----------
         qc_method : callable
             The query compiler method to call.
-        default_func : callable
-            The function to call if we need to default to pandas.
-        drop : bool, default: True
-            Whether or not to the grouping columns should be dropped on this operation.
-        numeric_only : bool, default: True
+        numeric_only : bool, default: None
             True for numeric only computations, False otherwise.
+        agg_args : list-like, optional
+            Positional arguments to pass to the aggregation function.
+        agg_kwargs : dict-like, optional
+            Keyword arguments to pass to the aggregation function.
         **kwargs : dict
-            The keyword arguments to be passed to the calling function.
+            Keyword arguments to pass to the specified query compiler's method.
 
         Returns
         -------
         DataFrame or Series
             Returns the same type as `self._df`.
         """
-        if self._axis != 0:
-            return self._default_to_pandas(default_func, **kwargs)
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if not self._is_multi_by and drop and self._drop and self._as_index:
-            groupby_qc = self._query_compiler.drop(columns=self._by.columns)
+        agg_args = tuple() if agg_args is None else agg_args
+        agg_kwargs = dict() if agg_kwargs is None else agg_kwargs
+
+        if numeric_only is None:
+            # pandas behaviour: if `numeric_only` wasn't explicitly specified then
+            # the parameter is considered to be `False` if there are no numeric types
+            # in the frame and `True` otherwise.
+            numeric_only = any(
+                is_numeric_dtype(dtype) for dtype in self._query_compiler.dtypes
+            )
+
+        if numeric_only and self.ndim == 2:
+            by_cols = self._internal_by
+            mask_cols = [
+                col
+                for col, dtype in self._df.dtypes.items()
```

Comment:
what's the difference between accessing `self._query_compiler.dtypes` and `self._df.dtypes`?..

Reply:
`self._df` could be a Series, whose `dtypes` property literally stores a `dtype` instead of a list of `dtypes` as DataFrames do. `query_compiler.dtypes` always stores a list of dtypes, that's why it's important to access QC's dtypes rather than `self._df` when iterating on them.

The picked line hasn't caused any problems because it's under `if self._df.ndim == 2` condition, but for the safety I replaced it with the `self._query_compiler.dtypes`.